src/common/utils/use-timeout/index.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback, useRef } from 'react';
+
+export const useTimeout = (delay: number) => {
+  const timeoutId = useRef<NodeJS.Timeout>();
+
+  const schedule = useCallback(
+    (callback: () => void) => {
+      if (timeoutId.current) {
+        clearTimeout(timeoutId.current);
+      }
+
+      timeoutId.current = setTimeout(callback, delay);
+    },
+    [delay]
+  );
+
+  const cancel = useCallback(() => {
```

Comment:
What's the purpose of `useCallback` here? Is it to ensure `clearTimeout(timeoutId.current)` runs once only regardless how many times `cancel` get called?